File: C/2129-capitalize-the-title/main.go
package main

import (
	"fmt"
	"strings"
)

func capitalizeTitle(title string) string {
	arr := strings.Split(title, " ")
	buf := []string{}
	for _, v := range arr {
		if len(v) <= 2 {
			buf = append(buf, strings.ToLower(v))
		} else {
			buf = append(buf, strings.Title(strings.ToLower(v)))
		}
	}
	return strings.Join(buf, " ")
}

func main() {
	fmt.Println()
}

File: C/2129-capitalize-the-title/main.rs
struct Solution {}

impl Solution {
  pub fn capitalize_title(title: String) -> String {
    title
      .split(" ")
      .map(|v| match v.len() {
        1 => v.to_lowercase(),
        2 => v.to_lowercase(),
        _ => {
          let v = v.to_lowercase();
          let mut c = v.chars();
          match c.next() {
            None => String::new(),
            Some(f) => f.to_uppercase().collect::<String>() + c.as_str(),
          }
        }
      })
      .collect::<Vec<String>>()
      .join(" ")
  }
}
